src/controllers/water-level-changes.js

```diff
@@ -0,0 +1,132 @@
+const request = require('request-promise-native')
+const uuid = require('uuid/v4')
+
+const TIMEZONE_OFFSET_MILLIS = 360 * 60 * 1000 // Timezone offset for Edmonton to UTC
+const KEY = 'water-level-changes' // Unique key for dataset storage
+const URL = 'https://data.edmonton.ca/resource/cnsu-iagr.json' // The Socrata api endpoint
+const QUERY_BASE = `${URL}?$query=`
+
+/**
+ * Water Level Changes (North Saskatchewan River at Edmonton)
+ */
+module.exports = async function(req, res) {
+  console.log('Water Level Changes Controller')
+  let storedData = await req.cache.getLatest(KEY)
+  let storedUpdated
+  if (storedData) storedUpdated = storedData.last_updated
+
+  let getUpdatedQuery = `${QUERY_BASE}
+    SELECT date_and_time, water_level_m
+    WHERE station_number = '05DF001'
+    ORDER BY date_and_time DESC
+    LIMIT 1`
+  let updatedData
+  let recentUpdated
+  let recentWaterLevel
+  try {
+    let rawJsonUpdated = await request(getUpdatedQuery)
+    updatedData = JSON.parse(rawJsonUpdated)
+    recentUpdated = updatedData[0].date_and_time
+    recentWaterLevel = updatedData[0].water_level_m
+  } catch (e) {
+    e.code = 500
+    throw e
+  }
+
+  let responseValues
+  if (storedUpdated && new Date(storedUpdated) >= new Date(recentUpdated)) {
+    console.log('No new updates')
+    responseValues = await req.cache.getAll(KEY, req.body['limit'])
+  } else {
```

Comment:
Nit: split else off next line. Does our linting allow this?

Reply:
Prettier formats the 'else' on the same line

Reply:
Lint also does not allow the 'else' on the next line

Reply:
Interesting. Leave this for now and I'll get it in #94.